mirage/client/dns_client_mirage.ml

```diff
@@ -134,9 +151,19 @@ module Make (R : Mirage_random.S) (T : Mirage_time.S) (M : Mirage_clock.MCLOCK)
       Lwt_condition.wait t.timer_condition >>= fun () ->
       loop ()
 
+    let udp_port = 12345
```

Comment:
TODO: should this be randomized? We'll need UDP listeners for each port (and ensure that either each is used only once or that unlisten is not done always (if there are two packets from port = x, only after the last packet sent the unlisten may be done)

Reply:
[RFC5452](https://www.rfc-editor.org/rfc/rfc5452) has a bunch of advice on this. It's a little scary to use a fixed source port for making DNS requests without DNSSEC verification. Spoofing responses is just so much easier if you can spam them to the same port.

Reply:
yes. refering to all of dan kaminsky work. that's why I put that comment here to remind ourselves to adjust the code before pressing merge and release ;)

Reply:
@haesbaert @avsm please check https://github.com/mirage/ocaml-dns/pull/322/commits/aa7458ad7a697f5a569e17b5abb5e4fdbd28a110 that should fix your reservations.